### DISCUSSION TITLE
New DEB package repository infrastructure.

### DISCUSSION THREAD

**Discussion by Ferroin** [netdata/netdata#14300] [Announcements]
As of 2023-01-20, Netdata has begun migration to new, self-hosted, infrastructure for our official DEB package repositories.  This new infrastructure provides a number of benefits, including edge-caching provided by CloudFlare, archive signing of packages, greatly improved support for local mirrors or caches of our DEB repositories.

If you are using native packages that were installed through our kickstart script, the next time you run updates you should see an update for either `netdata-repo` or `netdata-repo-edge` (depending on whether you are using stable packages or nightlies) to version 2. If you do see this (or are already on version 2 or newer of this package), you don’t need to do anything.

If you instead manually set up the repositories, or used PackageCloud’s install script to do so, please see the end of this post for information about how to configure the new repositories.

If you encounter any issues with the new infrastructure, please don’t hesitate to open an issue on GitHub about it. We will be treating such issues as high-priority bugs going forwards.

We will continue to publish packages to PackageCloud as well for the foreseeable future, but users are encouraged to switch to the new infrastructure as soon as possible.

---

Netdata’s official DEB repositories are hosted at http://repo.netdata.cloud/repos. We provide four groups of
repositories at that top level:

- `stable`: Contains packages for stable releases of the Netdata Agent.
- `edge`: Contains packages for nightly builds of the Netdata Agent.
- `repoconfig`: Provides packages that set up configuration files for using the other repositories.
- `devel`: Is used for one-off development builds of the Netdata Agent, and can simply be ignored by users.

Within each top level group of repositories, there are directories for each supported group of distributions:

- `debian`: Is for Debian Linux and binary compatible distros.
- `ubuntu`: Is for Ubuntu Linux and binary compatible distros.

Under each of these directories is a directory for each supported release, corresponding to the release codename.

These repositories are set up as what Debian calls ‘flat repositories’, and are available via both HTTP and HTTPS.

As a result of this structure, the required APT sources entry for stable packages for Debian 11 (Bullseye) is:

```
deb http://repo.netdata.cloud/repos/stable/debian/ bullseye/
```

Note the `/` at the end of the codename, and the space between the URL and the codename, both of these are required for the repository to be processed correctly.

Our DEB packages and repository metadata are signed using a GPG key with a user name of ‘Netdatabot’. The
current key fingerprint is `6588FDD7B14721FE7C3115E6F9177B5265F56346`. The associated public key can be fetched from
`https://repo.netdata.cloud/netdatabot.gpg.key`.

If you are explicitly configuring a system to use our repositories, the recommended setup is to download the
appropriate repository configuration package from https://repo.netdata.cloud/repos/repoconfig and install it

**Comment by alexisLefebvre** [comment]
Could you please explain how the new infra can be “self-hosted” and “provided by CloudFlare” too?

**Reply by Ferroin** [reply]
The new setup is using CloudFlare for edge-caching, but we’re the ones actually hosting the repository infrastructure. We could, in theory, completely remove CloudFlare from the setup with a single click and nobody would see any difference other than possibly being a bit slower.

**Reply by alexisLefebvre** [reply]
Thanks for your answer, I understand.

**Comment by Jay2k1** [comment]
The [docs](https://learn.netdata.cloud/docs/netdata-agent/installation/linux/native-linux-distribution-packages) still link to `repository.netdata.cloud` which doesn't work (https://repository.netdata.cloud/repos/repoconfig/ 404. https://repository.netdata.cloud/repos/ 404), however the link mentioned in this topic does work fine.
